Review user guide (#979)
ansible/receptor: docs/source/user_guide/configuration_options.rst

==============================
Receptor Configuration Options
==============================

---------------------
Receptor command line
---------------------

> Command line arguments have the form: receptor [--<action> [<param>=<value> ...] ...]

Reviewer: ```suggestion
Command line arguments use the following syntax: ``receptor [--<action> [<param>=<value> ...] ...]``
```

Author: all changes incorporated

> <action> are listed below.  When an action has parameters, they are listed in their own section.

Reviewer: ```suggestion
The possible options for ``<action>`` are listed below. Parameters for actions are listed in their own section.
```

-------------

> Miscellaneous

Reviewer 2: Miscellaneous to me seems misleading as it contains the most required param `node` 
Perhaps renaming to something like "Persistent Flags"

-------------

.. list-table:: Miscellaneous
    :header-rows: 1
    :widths: auto

    * - Action
      - Description

>     * - --bash-completion

Reviewer: ```suggestion
    * - ``--bash-completion``
```

Reviewer: I suggest wrapping all command line options and parameters with backticks.

>       - Generate a completion script for the bash shell

Reviewer: ```suggestion
      - Generates a completion script for the Bash shell.
```

    * - --config <filename>

>       - Load additional config options from a YAML file

Reviewer: ```suggestion
      - Loads additional configuration options from a YAML file.
```

    * - --control-service

>       - Run a control service

Reviewer: ```suggestion
      - Runs a control service.
```

Reviewer: Is the control service a special type of node? Should we say "control service node"?

Author: The control service a service that may run on any type of Receptor node.  I also changed the help text

    * - --help

>       - Show this help

Reviewer: ```suggestion
      - Displays this help text.
```

Author: Similarly for receptor-tester.sh

    * - --local-only

>       - Run a self-contained node with no backends

Reviewer: ```suggestion
      - Runs a self-contained node without any backend services.
```

Reviewer: "With no backend" - does this mean with no persistence? If so, I would prefer to spell out what that means. Maybe just describe it as ephemeral.

Author: It means it runs without any backend (below the mesh - tcp/udp/websocket) processes.  So it can only talk to itself.

How about "Runs a self-contained node without any backend services." ?

Reviewer: Sounds good to me. Thanks for explaining!

    * - --log-level

>       - Set specific log level output

Reviewer: ```suggestion
      - Specifies the verbosity level for logging output.
```

Reviewer: Where do you find the possible levels? What is the default?

Author: What about "Specifies the verbosity level for logging output." ?

The log levels are below along with the default.  Do we need a reference?

Reviewer: Yeah, that suggestion sounds better. You could say something like "See the available verbosity levels in the following section for more information." as a reference but I would avoid any sort of linking (if that is even possible here.)

    * - --node

>       - Node configuration of this instance (required)

Reviewer: ```suggestion
      - Specifies the node configuration of this instance. This parameter is required.
```

    * - --trace

>       - Enables packet tracing output

Reviewer: ```suggestion
      - Enables packet tracing output.
```

    * - --version

>       - Show the Receptor version

Reviewer: ```suggestion
      - Displays the Receptor version.
```

Reviewer: Aha! So it is "Receptor". This should change globally when referring to the project/product name.

Author: This came from the code. (help)  I also updated the text there.

>       - Specifies the node configuration of this instance.  THis parameter is required.

Reviewer 2: THis should be This

^^^^^^^^^^^^^^^
Bash completion
^^^^^^^^^^^^^^^

> Run ". <(receptor --bash-completion)" to activate now

Reviewer: ```suggestion
Run ``. <(receptor --bash-completion)`` to activate now
```

Reviewer: What do you mean by "activate now"?

Author: Clarified text.

^^^^^^^^^^^^^^^
Control Service
^^^^^^^^^^^^^^^

.. list-table:: Control Service
    :header-rows: 1
    :widths: auto

    * - Parameter
      -  Description

>       -  Default

Reviewer: ```suggestion
      -  Default value
```

Author: Changed for each table

    * - filename=<string>

>       - Filename of local Unix socket to bind to the service

Reviewer: ```suggestion
      - Specifies the filename of a local Unix socket to bind to the service.
```

Author: also changed in source

>       -

Reviewer: ```suggestion
      - No default value.
```

Author: Added to all instances where no default value was specified

    * - ``permissions=<int>``
      - Socket file permissions
      - 0600
    * - ``service=<string>``
      - Receptor service name to listen on
      - control
    * - ``tls=<string>``
      - Name of TLS server config for the Receptor listener
      - No default value.
    * - ``tcplisten=<string>``
      - Local TCP port or host:port to bind to the control service
      - No default value.
    * - ``tcptls=<string>``
      - Name of TLS server config for the TCP listener
      - No default value.

^^^^^^^^^
Log level
^^^^^^^^^

.. list-table:: Log Level
    :header-rows: 1
    :widths: auto

    * - Parameter
      - Description
      - Default value
    * - ``level=<string>``
      - Log level: Error, Warning, Info or Debug
      - Error

^^^^
Node
^^^^

.. list-table:: Node
    :header-rows: 1
    :widths: auto

    * - Parameter
      - Description
      - Default value
    * - ``id=<string>``
      - Node ID
      - local hostname
    * - ``datadir=<string>``
      - Directory in which to store node data

>       - No default value.

Reviewer 2: default is `/tmp/receptor`

Reviewer 2: This will change with PR #951

Author: PR 951 needs to address both places (doc and code)

Reviewer 2: the default right now without pr #951 is `/tmp/receptor/` 
The second comment was just FYI

    * - ``firewallrules=<JSON list of JSON dict of JSON data to JSON data>``

>       -  Firewall Rules (see documentation for syntax)

Reviewer 2: (see documentation for syntax), maybe link here

      - No default value.
    * - ``maxidleconnectiontimeout=<string>``
      - Max duration with no traffic before a backend connection is timed out and refreshed
      - No default value.

------------------------------------------
Configure resources used by other commands
------------------------------------------

.. list-table:: Configure resources used by other commands
    :header-rows: 1
    :widths: auto

    * - Action
      - Description
    * - ``--tls-client``
      - Define a TLS client configuration
    * - ``--tls-server``
      - Define a TLS server configuration

^^^^^^^^^^
TLS Client
^^^^^^^^^^

.. list-table:: TLS Client
    :header-rows: 1
    :widths: auto

    * - Parameter
      - Description
      - Default value
    * - ``cert=<string>``

>       - Client certificate filename

Reviewer 2: required

      - No default value.
    * - ``insecureskipverify=<bool>``
      - Accept any server cert
      - false
    * - ``key=<string>``

>       - Client private key filename

Reviewer 2: required
